frontend/actions/tagBlue.js

```diff
@@ -0,0 +1,31 @@
+// actions.js
+import { createSlice } from '@reduxjs/toolkit'
+export const BlueTag = createSlice({
+  name: 'bluetag',
+  initialState: {
+    value: [], // list of string: postids 
+  },
+  reducers: {
+    blue: (state, action) => {
+		// Redux Toolkit allows us to write "mutating" logic in reducers. It
+		// doesn't actually mutate the state because it uses the Immer library,
+			// which detects changes to a "draft state" and produces a brand new
+			// immutable state based off those changes
+	    var id = action.payload // value of postid
+			var temp = [...state.value]
+			// add to workspace
+		var i = temp.indexOf(id)
+		if (i > -1) {
+		  temp.splice(i, 1)
```

Comment:
if you wanted to keep using this logic, then I would suggest something like:

{id: "postid", colour: "red"}

as the item that you're adding. Note that object equivalence is not always easy, so you may need to pass a function to indexof OR just define your own version of indexOf